librepo/handle.c

```diff
@@ -264,6 +266,33 @@ lr_handle_remote_sources_changed(LrHandle *handle, LrChangedRemoteSource type)
     }
 }
 
+struct callback_data {
+   GMainLoop *loop; 
+   long input_seconds;
+   time_t begin;
+   GNetworkMonitor *monitor;
+   GSocketConnectable *connectable; 
+   GCancellable *cancellable;
+};
+
+gboolean timeout_callback(gpointer data)
+{ 
+  struct callback_data *dt = (struct callback_data*)data;
+   if(!g_network_monitor_can_reach(dt->monitor, dt->connectable, dt->cancellable, NULL) &&
+      time(NULL) - dt->begin <= dt->input_seconds){
```

Comment:
`time` operates in terms of wall clock time - this means it can change if e.g. the system clock is out of date and resynchronizes.

Further, when the system suspends (e.g. closing laptop lid) then the values can make *large* leaps.

We should use `g_get_monotonic_time()` here instead.  A bit more about [monotonic clocks](https://itnext.io/as-a-software-developer-why-should-you-care-about-the-monotonic-clock-7d9c8533595c).